system/Commands/Translation/LocalizationFinder.php

```diff
@@ -0,0 +1,339 @@
+        $currentDir         = APPPATH;
+        $this->languagePath = $currentDir . 'Language';
+        $tableRows          = [];
+        $languageFoundKeys  = [];
+        $countFiles         = 0;
+        $countNewKeys       = 0;
+
+        if (ENVIRONMENT === 'testing') {
+            $currentDir         = SUPPORTPATH;
+            $this->languagePath = $currentDir . 'Language/';
+        }
+
+        if (is_string($cliOptionLocale)) {
+            if (! in_array($cliOptionLocale, config(App::class)->supportedLocales, true)) {
+                CLI::error('Error: Supported locales ' . implode(', ', config(App::class)->supportedLocales));
+
+                return -1;
+            }
+
+            $currentLocale = $cliOptionLocale;
+        }
+
+        if (is_string($cliOptionDir)) {
+            $tempCurrentDir = realpath($currentDir . $cliOptionDir);
+
+            if (false === $tempCurrentDir) {
+                CLI::error('Error: Dir must be located in ' . $currentDir);
+
+                return -1;
+            }
+
+            if (str_starts_with($tempCurrentDir, $this->languagePath)) {
+                CLI::error('Error: Dir ' . $this->languagePath . ' restricted to scan.');
+
+                return -1;
+            }
+
+            $currentDir = $tempCurrentDir;
+        }
+
+        $iterator = new RecursiveIteratorIterator(new RecursiveDirectoryIterator($currentDir));
+        $this->writeIsVerbose('Current locale: ' . $currentLocale);
+        $this->writeIsVerbose('Find phrases in ' . $currentDir . ' folder...');
+
+        /**
+         * @var SplFileInfo $file
+         */
+        foreach ($iterator as $file) {
+            if ($this->isIgnoredFile($file)) {
+                continue;
+            }
+
+            $this->writeIsVerbose('File found: ' . mb_substr($file->getRealPath(), mb_strlen(APPPATH)));
+            $countFiles++;
+            $languageFoundKeys = array_replace_recursive($this->findTranslationsInFile($file), $languageFoundKeys);
+        }
+
+        ksort($languageFoundKeys);
+        $languageDiff        = [];
+        $languageFoundGroups = array_unique(array_keys($languageFoundKeys));
+
+        foreach ($languageFoundGroups as $langFileName) {
+            $languageStoredKeys = [];
+            $languageFilePath   = $this->languagePath . DIRECTORY_SEPARATOR . $currentLocale . DIRECTORY_SEPARATOR . $langFileName . '.php';
+
+            if (is_file($languageFilePath)) {
+                /**
+                 * Load old localization
+                 */
+                $languageStoredKeys = require $languageFilePath;
+            }
+
+            $languageDiff = $this->arrayDiffRecursive($languageFoundKeys[$langFileName], $languageStoredKeys);
+            $countNewKeys += $this->arrayCountRecursive($languageDiff);
+
+            if ($cliOptionShowNew) {
+                $tableRows = array_merge($this->arrayToTableRows($langFileName, $languageDiff), $tableRows);
```

Comment:
I would lke use

```suggestion
                $tableRows = [$this->arrayToTableRows($langFileName, $languageDiff), $tableRows];
```

Ref : https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md#arraymergeofnonarraystosimplearrayrector

Reply:
No. Read the instructions carefully. 
You are trying to merge two arrays, and the documentation talks about strings. This will result in an error.
+ I applied the Rector before the commit